test/test_torch.py

```diff
@@ -11882,6 +11882,50 @@ def test_ceil_out_mismatch(self, device):
         b = torch.randn(1, device=device)
         self.assertRaises(RuntimeError, lambda: torch.ceil(a, out=b))
 
+    # test_<func>_type_upcasting are sample test functions (to test implicit dtype promotion) for sample functions of each category
+    # Check https://github.com/pytorch/pytorch/pull/33322 for more details
+    @dtypes(torch.int8, torch.int16, torch.int32, torch.int64, torch.bool)
+    def test_ceil_type_upcasting(self, device, dtype):
+        x = torch.tensor([2.3], dtype=dtype, device=device)
+        out = torch.ceil(x)
+
+        if(dtype == torch.int8):
+            if(out.device.type == 'cpu'):
+                self.assertIs(out.dtype, torch.float32)
+            else:
+                self.assertIs(out.dtype, torch.float16)
+        elif(dtype == torch.int16):
+            self.assertIs(out.dtype, torch.float32)
+        elif(dtype == torch.int32):
+            self.assertIs(out.dtype, torch.float64)
+        elif(dtype == torch.int64):
+            self.assertIs(out.dtype, torch.float64)
+        elif(dtype == torch.bool):
+            if (out.device.type == 'cpu'):
+                self.assertIs(out.dtype, torch.float32)
+            else:
+                self.assertIs(out.dtype, torch.float16)
+
+    @dtypes(torch.int8, torch.int16, torch.int32, torch.int64, torch.bool)
+    def test_angle_type_upcasting(self, device, dtype):
```

Comment:
To make this generic, have a list of op name strings, and do something like
```def test_type_1(self):
        type_1_list = ["angle", ...]
        for opname in type_1_list:
            getattr(torch, opname)(arg)
```

Reply:
Thanks, @mcarilli. I've added the generic test functions in the recent commits.